src/linter/root.go

```diff
@@ -69,10 +69,12 @@ type rootWalker struct {
 	// name matches the pattern and @linter disable was encountered
 
 	// strictTypes is true if file contains `declare(strict_types=1)`.
-	strictTypes bool
-	strictMixed bool
+	strictTypes    bool
+	strictMixed    bool
+	declareSection bool
```

Comment:
I propose to do it differently. We have an `ir.Root` node that describes the entire file. Essentially, we can check if there is an `ir.DeclareStmt` at its level, and if there is, show an error.

see `block.go`